### ISSUE TITLE
fab install fails when seed repo has newer pinned changeset than on disk in dependent repo

### ISSUE THREAD

**Issue by shadowspawn** [JohnRGee/forest-arborist#14]
# Bug Report

## Summary

Running `fab install` will fail when a pinned dependency is updated in the seed repo, but the repo already on disk is stale and does not have the referenced changeset.

This is in contrast to running `fab install` after adding a repo, which will clone the repo if required.

@kitadamsadi @rgroenendal

## Steps to Reproduce

1. two sandpits with a forest with a dependent repo using a locked changeset
2. add a new changeset to dependent repo in sandpit A
3. change pinned changeset in seed repo of sandpit A
4. push both repos to server
5. in sandpit B, pull on seed repo and _not_ on dependent repos
6. in sandpit B, run `fab install`

## Expected Behaviour

`fab install` is the goto tool for catching up with changes in manifest and dependent repos. Expect forest to be updated as necessary.

## Actual Behaviour

```sh
$ fab install

Reinitialised marker file at root of forest: .fab-root.json

# deps/Boost: checkout pinned revision
deps/Boost: git -c advice.detachedHead\=false checkout b9be6c4d84d13c92066a36a3596e4f2dc6581773
fatal: reference is not a tree: b9be6c4d84d13c92066a36a3596e4f2dc6581773
caught exception with message Command failed: git -c advice.detachedHead=false checkout b9be6c4d84d13c92066a36a3596e4f2dc6581773
```

**Comment by shadowspawn** [comment]
In a forest with only one DVCS could run something like `fab git fetch` as a quick work-around to update repos, but probably not familiar to casual users of command-line (those using GUI tools for most operations).

**Comment by kitadamsadi** [comment]
Could we put that in our npm install scripts?

**Comment by shadowspawn** [comment]
Not quite sure which scripts, but likely yes. (And Kit confirmed it worked.)

**Comment by shadowspawn** [comment]
Fixed in v6.0.2 by doing an aggressive refresh of existing dependent repos, `git fetch` and `hg pull`. Seems reasonable for an install.
